### DISCUSSION TITLE
Clustering using ClusterData results in AttributeError

### DISCUSSION THREAD

**Discussion by arunraja-hub** [pyg-team/pytorch_geometric#8961] [Q&A]
I just tried clustering qm9 and I got the following error.
Items is not an attribute in the Qm9 dataset class: https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/datasets/qm9.html#QM9

```
cluster = loader.ClusterData(qm9_data, 5)
```

`AttributeError: 'qm9' object has no attribute 'items'
`

@rusty1s

**Comment by rusty1s** [comment]
This is expected. `ClusterData` needs to receive a single graph as input while `QM9` holds a set of graphs (for which there is no point in partitioning them in the first place).
